cgp/genome.py

```diff
@@ -388,6 +395,21 @@ def set_expression_for_output(
         self.change_address_gene_of_output_node(
             new_address=last_inserted_node, output_node_idx=output_node_idx
         )
+        try:
+            import sympy
+
+            if target_expression is not None:
+                if self._n_outputs > 1:
+                    output_as_sympy = CartesianGraph(self).to_sympy()[output_node_idx]
+                else:
+                    output_as_sympy = CartesianGraph(self).to_sympy()
+
+                target_expression_as_sympy = sympy.parse_expr(target_expression)
+                if not output_as_sympy == target_expression_as_sympy:
+                    raise ValueError("expression of output and target expression do not match")
+
+        except ModuleNotFoundError:
+            raise Warning("Sympy not available, can not compare written output to target")
```

Comment:
```suggestion
            raise ModuleNotFoundError("No module named 'sympy' (extra requirement)")
```
as mentioned above, i would raise an exception with our standard error message. you can possibly extend this with more detailed info, e.g., "can not check output expression. No module..."